crates/ruff_linter/src/rules/flake8_implicit_str_concat/rules/explicit.rs

```diff
@@ -34,46 +34,71 @@ use crate::settings::LinterSettings;
         "Explicitly concatenated string should be implicitly concatenated".to_string()
     }
+
+    fn fix_title(&self) -> String {
+        "Remove redundant '+' operator to implicitly concatenate".to_string()
+    }
 }
 
 /// ISC003
-pub(crate) fn explicit(
-    expr: &Expr,
-    locator: &Locator,
-    settings: &LinterSettings,
-) -> Option<Diagnostic> {
+pub(crate) fn explicit(expr: &Expr, checker: &Checker) -> Option<Diagnostic> {
     // If the user sets `allow-multiline` to `false`, then we should allow explicitly concatenated
     // strings that span multiple lines even if this rule is enabled. Otherwise, there's no way
     // for the user to write multiline strings, and that setting is "more explicit" than this rule
     // being enabled.
-    if !settings.flake8_implicit_str_concat.allow_multiline {
+    if !checker.settings.flake8_implicit_str_concat.allow_multiline {
         return None;
     }
 
-    if let Expr::BinOp(ast::ExprBinOp {
-        left,
-        op,
-        right,
-        range,
-    }) = expr
-    {
-        if matches!(op, Operator::Add) {
-            if matches!(
-                left.as_ref(),
-                Expr::FString(_) | Expr::StringLiteral(_) | Expr::BytesLiteral(_)
-            ) && matches!(
-                right.as_ref(),
-                Expr::FString(_) | Expr::StringLiteral(_) | Expr::BytesLiteral(_)
-            ) && locator.contains_line_break(*range)
-            {
-                return Some(Diagnostic::new(ExplicitStringConcatenation, expr.range()));
+    if let Expr::BinOp(bin_op) = expr {
+        if let ast::ExprBinOp {
+            left,
+            right,
+            range,
+            op: Operator::Add,
+        } = bin_op
+        {
+            let concatable = matches!(
+                (left.as_ref(), right.as_ref()),
+                (
+                    Expr::StringLiteral(_) | Expr::FString(_),
+                    Expr::StringLiteral(_) | Expr::FString(_)
+                ) | (Expr::BytesLiteral(_), Expr::BytesLiteral(_))
+            );
+            if concatable && checker.locator().contains_line_break(*range) {
+                let mut diagnostic = Diagnostic::new(ExplicitStringConcatenation, expr.range());
+                diagnostic.set_fix(generate_fix(checker, bin_op));
+                return Some(diagnostic);
             }
         }
     }
     None
 }
+
+fn generate_fix(checker: &Checker, expr_bin_op: &ast::ExprBinOp) -> Fix {
+    let ast::ExprBinOp { left, right, .. } = expr_bin_op;
+    let operator_range = TextRange::new(left.end(), right.start());
+    let operator_text = checker.locator().slice(operator_range);
+
+    let plus_pos = operator_text.find('+').unwrap();
+
+    let (before, after) = operator_text.split_at(plus_pos);
+    let after = &after[1..]; // Ignore `+` operator
+
+    // With `+` on first line a newline isn't in before; trim excess whitespace
+    let before = if before.contains('\n') || before.contains('\r') {
```

Comment:
Can we add some more multiline tests, demonstrating that this works as expected. E.g. 

```py
_ = (
    "start" +
    variable + # comment
    "end"
)
```

```py
_ = (
    "start" +
    variable 
		# leading comment
    + "end"
)
```

I think what I would do instead of using `contains` is to call `trim_end_matches(' ')`. This also avoids trimming any linefeed characters.

Reply:
`trim_end_matches(|c| c == '\t' || c == '\f' || c == ' ')` would catch all the non-newline spaces.